api/src/main/java/io/cloudevents/format/EventFormatProvider.java

```diff
@@ -0,0 +1,34 @@
+package io.cloudevents.format;
+
+import io.cloudevents.format.json.JsonFormat;
+
+import java.util.HashMap;
+
+public final class EventFormatProvider {
+
+    private static class SingletonContainer {
+        private final static EventFormatProvider INSTANCE = new EventFormatProvider();
+    }
+
+    public static EventFormatProvider getInstance() {
+        return EventFormatProvider.SingletonContainer.INSTANCE;
+    }
+
+    private HashMap<String, EventFormat> formats;
```

Comment:
`String` as a key is hard to work with, given that it can be something like "application/json; charset=utf-8" (the current version of SDK's HTTP support fails because of that "; charset=utf-8", btw)

Reply:
I think a correct usage should invoke `resolveFormat` with a parsed content type header (so without passing `charset`)